autopaho/auto.go

```diff
@@ -554,10 +554,20 @@ connectionLoop:
 				if errors.Is(err, queue.ErrEmpty) {
 					c.debug.Println("everything in queue transmitted")
 					continue queueLoop
+				} else if err != nil {
+					// if Peek() keeps returning errors, we will loop forever.
+					// see https://github.com/eclipse/paho.golang/issues/234
+					c.errors.Printf("error retrieving queue entry: %s", err)
+					time.Sleep(1 * time.Second)
```

Comment:
I've put a hardcoded 1 second delay any time we are retrying the same publish. Open to feedback on a different approach.